src/coll_patterns/sra_knomial.h

```diff
@@ -265,6 +265,18 @@ static inline void ucc_kn_ag_pattern_next_iter(ucc_knomial_pattern_t *p)
     }
 }
 
+static inline void ucc_kn_rs_pattern_init(ucc_rank_t size, ucc_rank_t rank,
+                                          ucc_kn_radix_t radix, size_t count,
+                                          ucc_knomial_pattern_t *p)
+{
+    ucc_knomial_pattern_init_backward(size, rank, radix, p);
+    p->type              = KN_PATTERN_REDUCE_SCATTER;
+    p->count             = count;
+    p->block_size_counts = count;
+    p->block_size        = size - p->n_extra;
+    p->block_offset      = 0;
+}
+
 static inline void ucc_kn_rsx_pattern_init(ucc_rank_t size, ucc_rank_t rank,
```

Comment:
what does suffix "x" mean?

Reply:
rsx - reduce scater part of allreduce
rs - reduce scatter block (in mpi terminology)
rsv - reduce scatter (in mpi terminology)